File: GloceryMgtFinal/GloceryMgt/src/app/_models/report.ts
export class Report {

    constructor( private _reportId: Number , private _userName : String ,private _reportName : String , private _reportData : String , private _reportTime : Date  )
    {
        this.reportId = _reportId ;
        this.userName = _userName;
        this.reportName = _reportName;
        this.reportData = _reportData;
        this.reportTime = _reportTime;

    }
    reportId : Number;
    userName : String;
    reportName : String;
    reportData : String;
    reportTime : Date;

}

File: WebApi/Model/Report.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;

namespace WebApi.Models
{
    public class Report
    {
        [Key]
        public int ReportId { get; set; }
        public string UserName { get; set; }
        public string  ReportName { get; set; }

        public string ReportData { get; set; }

        public DateTime? ReportTime { get; set; } = DateTime.Now;

    }
}